manufacturing-client/src/main.rs

```diff
@@ -401,16 +417,19 @@ async fn main() -> Result<()> {
             // and it might request MACAddress as the mfg_string_type. What it cannot do
             // is select the iface for the client, so we must set it ahead of time.
             // Thereby, we are just getting this value if provided, hiding errors.
-            if env::var("DI_MFG_STRING_TYPE_MAC_IFACE").is_ok() {
-                iface = Some(env::var("DI_MFG_STRING_TYPE_MAC_IFACE").unwrap());
+            if let Ok(iface_var) = env::var("DI_MFG_STRING_TYPE_MAC_IFACE") {
+                iface = Some(iface_var);
             }
             (keyref, mfg_string_type) = perform_diun(&mut client, diun_pub_key_verification)
                 .await
                 .context("Error performing DIUN")?;
             // once diun has been performed we can error ahead of time if the
             // server has requested MACaddress but we haven't set an iface.
```

Comment:
you can delete/rewrite this code comment since it does not apply now after the default iface feature